File: competitive_programming/codechef/long_challenge/JUN21C/CHFHEIST.cpp
#include <iostream>
using namespace std;

int main() {
	long int t, D, d, P, Q, n, sumQ, ans;
	
	scanf("%ld", &t);
	while(t--){
	    scanf("%ld %ld %ld %ld", &D, &d, &P, &Q);
	    n = D/d - 1;
	    sumQ = (n*(n+1))/2;
	    ans = (P*D) + (sumQ*Q*d) + ((D%d)*(n+1)*Q);
	    printf("%ld\n", ans);
	}
	return 0;
}

File: competitive_programming/codechef/long_challenge/JUN21C/CHFHEIST.py
# cook your dish here
for _ in range(int(input())):
    D,d,P,Q = map(int, input().split())
    
    n = (D//d) - 1
    sumQ = (n*(n+1))/2
    
    ans = (P*D) + (sumQ*Q*d) + ((D%d)*(n+1)*Q)
    
    print(int(ans))
